packages/substrate-bindings/src/codecs/AccountId.ts

```diff
@@ -6,6 +6,8 @@ const SS58_PREFIX = new TextEncoder().encode("SS58PRE")
 
 const CHECKSUM_LENGTH = 2
 
+export type SS58String = string & { __SS58String: unknown }
```

Comment:
This is a branded type, right?
(I [read](https://egghead.io/blog/using-branded-types-in-typescript) about this last week)

Perhaps we could have a type util like
```
declare const __brand: unique symbol
type Brand<B> = { [__brand]: B }

export type Branded<T, B> = T & Brand<B>

export type SS58String = Branded<string, "SS58String">

Reply:
yeah... TBH I'm still not sure what the best way to go here is. I will probably add this type util later, for sure!